.travis.yml

```diff
@@ -0,0 +1,104 @@
+
+    # CMake
+    - sudo add-apt-repository --yes ppa:kalakris/cmake
+
+    # Update all the repositories
+    - sudo apt-get update -qq
+
+    # Mongo C Driver
+    - curl -O -L https://github.com/mongodb/mongo-c-driver/releases/download/${CDRIVER_VERSION}/mongo-c-driver-${CDRIVER_VERSION}.tar.gz
+
+    # Mongo C++ Driver
+    - git clone -b 'master' --single-branch https://github.com/mongodb/mongo-cxx-driver.git
+
+install:
+    # GCC
+    - if [ "$CXX" = "g++" ]; then sudo apt-get install -qq g++-6; export CXX="g++-6"; export CC="gcc-6"; export NO_ERROR_ON_MAYBE_UNINITIALIZED="-Wno-error=maybe-uninitialized"; fi
+
+    # Clang
+    - if [ "$CXX" == "clang++" ]; then sudo apt-get install --allow-unauthenticated -qq clang-3.8; export CXX="clang++-3.8"; export CC="clang-3.8"; fi
+
+    # CMake
+    - tar -zxf ${CMAKE_VERSION}.tar.gz -C build
+
+    # Install MongoDB Enterprise
+    - sudo apt-get install mongodb-enterprise-server
+
+    # Install CMake (Versions lower than 2.8.8 do not support object libraries)
+    - sudo apt-get install cmake
+
+    # Prepare pkg-config path
+    - export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/home/travis/install/lib/pkgconfig/
+
+    # Install Mongo C Driver
+    - tar -zxf mongo-c-driver-${CDRIVER_VERSION}.tar.gz
+    - pushd mongo-c-driver-${CDRIVER_VERSION}; ./configure --prefix=/home/travis/install; make; sudo make install; popd;
+
+    # Install Mongo C++ Driver
+    - pushd mongo-cxx-driver/build; ../../build/${CMAKE_BINARY} -DCMAKE_CXX_FLAGS="-Wno-ignored-attributes" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/home/travis/install/ ..; make; sudo make install; popd; 
+
+before_script:
+    - $CC --version
+    - $CXX --version
+
+    # Build the odm and the tests
+    - cd build
+    - if [ "$CONFIG" == "Release" ]; then ${CMAKE_BINARY} -DCMAKE_BUILD_TYPE=$CONFIG -DCMAKE_MODULE_PATH=/home/travis/build/mongodb/mongo-cxx-odm/mongo-cxx-driver/cmake -DCMAKE_INSTALL_PREFIX=/home/travis/install -DCMAKE_C_FLAGS="-Wall -Wextra -Wno-attributes -Werror -Wno-error=missing-field-initializers ${NO_ERROR_ON_MAYBE_UNINITIALIZED}" -DCMAKE_CXX_FLAGS="-Wall -Wextra -Wno-attributes -Werror -Wno-error=missing-field-initializers ${NO_ERROR_ON_MAYBE_UNINITIALIZED}" ..; fi
+    - if [ "$CONFIG" == "Debug" ]; then ${CMAKE_BINARY} -DCMAKE_BUILD_TYPE=$CONFIG -DCMAKE_MODULE_PATH=/home/travis/build/mongodb/mongo-cxx-odm/mongo-cxx-driver/cmake -DCMAKE_INSTALL_PREFIX=/home/travis/install -DCMAKE_C_FLAGS="-Wall -Wextra -Wno-attributes -Werror -Wno-error=missing-field-initializers" -DCMAKE_CXX_FLAGS="-Wall -Wextra -Wno-attributes -Werror -Wno-error=missing-field-initializers" ..; fi
+
+script:
+    - make format-lint
+
+    - make all
+
+    # Run bson_mapper tests with catch
+    - ./src/bson_mapper/test/test_bson_mapper
+
+    # Run mongo_odm tests with catch
+    - ./src/mongo_odm/test/test_mongo_odm
+
+    # Install headers and libs for the examples
+    - sudo make install
```

Comment:
I would like to figure out how to do this without root. Our users should be able to install without sudo as long as they configured the directory properly, so let's get that working here too to be sure.